modules/desktop/profiles/applications.nix

```diff
@@ -4,21 +4,112 @@
 {
   config,
   lib,
+  pkgs,
   ...
 }: let
   cfg = config.ghaf.profiles.applications;
-in {
-  options.ghaf.profiles.applications = {
-    enable = lib.mkEnableOption "Some sample applications";
-    #TODO Create options to allow enabling individual apps
-  };
+in
+  with lib; {
+    options.ghaf.profiles.applications = {
+      enable = mkEnableOption "Some sample applications";
+      #TODO Create options to allow enabling individual apps
+      #weston.ini.nix mods needed
+      ivShMemServer = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Enable memory sharing between virtual machines";
+        };
+        memSize = mkOption {
+          type = lib.types.int;
+          default = 16;
+          description = mdDoc ''
+            Defines shared memory size
+          '';
+        };
+        vmCount = mkOption {
+          type = lib.types.int;
+          default = 6;
+          description = mdDoc ''
+            Defines maximum number of application VMs
+          '';
+        };
+        serverSocketPath = mkOption {
+          type = lib.types.str;
+          default = "/run/user/1000/memsocket-server.sock";
```

Comment:
Binding the path to `ghaf` `UID` can be error-prone. It might be an acceptable risk but the user-model is due to get changed to operator and admin model where we can't assume the operator user to have the same rights as it currently has. Depending on how nix creates the declared operator and admin users, the UID bound path can break things.

Reply:
I added the `uid` parameter to `options.ghaf.users.accounts`, as the same issue exists for nm-launcher.